osu.Game/Overlays/Settings/Sections/Maintenance/SystemFileImportComponent.cs

```diff
@@ -0,0 +1,51 @@
+// Copyright (c) ppy Pty Ltd <contact@ppy.sh>. Licensed under the MIT Licence.
+// See the LICENCE file in the repository root for full licence text.
+
+using System.Linq;
+using System.Threading.Tasks;
+using osu.Framework.Graphics;
+using osu.Framework.Platform;
+
+namespace osu.Game.Overlays.Settings.Sections.Maintenance
+{
+    public partial class SystemFileImportComponent : Component
+    {
+        private readonly OsuGameBase game;
+        private readonly GameHost host;
+
+        private ISystemFileSelector? selector;
+
+        public SystemFileImportComponent(OsuGameBase game, GameHost host)
+        {
+            this.game = game;
+            this.host = host;
+        }
+
+        protected override void LoadComplete()
+        {
+            base.LoadComplete();
+
+            selector = host.CreateSystemFileSelector(game.HandledExtensions.ToArray());
+
+            if (selector != null)
+                selector.Selected += f => Schedule(() => startImport(f.FullName));
```

Comment:
Scheduling a `Task.Run`? Any reason?

Reply:
Just a habit to schedule any action subscribed from an event of an external component inside a drawable class, but it doesn't matter that much anyway.